pkg/k8shandler/visualization.go

```diff
@@ -236,20 +240,29 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateOauthClient(oauthSecr
 				return fmt.Errorf("Failed to get %v oauthclient for %q: %v", oauthClient.Name, clusterRequest.cluster.Name, err)
 			}
 
+			//Check to see if we even need to update
+			//remove annotation to normalize
+			updatedAt, _ = current.GetObjectMeta().GetAnnotations()[annotationOauthSecretUpdatedAt]
+			delete(current.GetObjectMeta().GetAnnotations(), annotationOauthSecretUpdatedAt)
```

Comment:
instead of doing this could we not just check `RedirectUIRs` `Secret` and `ScopeRestrictions` ?

Reply:
Isn't it easier to simply delete the thing we know to be different?  Also, why do we copy oauthClient in the first place if the 'Get' call is going to load the object anyway?

Reply:
we don't need to copy it.... either copy or create an empty oauthclient object. the copy method is left over from the older sdk logic we used...

Reply:
right but if the updatedAt field is different we aren't going to do an update. we only care to check if those three fields i listed above are different... so lets just check those.